### ISSUE TITLE
`distinct enum` not supported (invalid indentation error)

### ISSUE THREAD

**Issue by awr1** [nim-lang/Nim#8604]
```nim
type Character = distinct enum
  fred, daphne, velma, shaggy, scooby
```
Output:
```
c:/nimtests $ nim c distinctenum.nim
Hint: used config file 'c:\Nim\config\nim.cfg' [Conf]
Hint: system [Processing]
Hint: distinctenum [Processing]
distinctenum.nim(2, 27) Error: invalid indentation
```
Nim already supports using `distinct` in this way with `distinct object` and `distinct tuple`. Workaround:
```nim
type
  Character = enum
    fred, daphne, velma, shaggy, scooby
  DistinctCharacter = distinct Character
```

**Comment by awr1** [comment]
Thinking about this a little more...this brings up a question of *"what is made useful by allowing `distinct enum`s?"* How would someone accomplish something similar to this statement:
```nim
var a = fred
```
I came across this via experimentation - I like to use distinct types to override the meaning of the `$` stringify proc (which may or may not be bad style) - but the more I think about it, the more I find the usage of this questionable...

**Comment by dom96** [comment]
I'm surprised that `distinct object`/`distinct tuple` is a thing and I am having a hard time figuring out a use case for it.

After reading your comments I also don't understand what the use for a ``distinct enum`` is...

**So IMO we should just remove `distinct object`/`distinct tuple` instead.**

**Comment by twetzel59** [comment]
@dom96 I agree, because ``distinct`` is designed to declare a type that differs **semantically** from the base type. Per the manual:
```
A distinct type is new type derived from a base type that is incompatible with its base type. In particular, it is an essential property of a distinct type that it does not imply a subtype relation between it and its base type.
```

``distinct object``, ``distinct tuple``, and said ``distinct enum`` are redundant, because if they were to work, they would declare a type incompatible with, well... *itself*.

@awr1 I find the ``$`` thing interesting, but I'm surprised that even works. Semantically that's sorta strange. Perhaps it would be more useful to provide a way to turn off generating those automatic implementations, e.g. ``proc `$``` for a type.

On second thought... yikes! Probably better not to break understood language behavior at the type declaration site.

**Comment by awr1** [comment]
I do think there should be a proper way to override `$` for a type. You can define it in a module, but the user-defined stringification proc will only work inside that module - unless you make the type `distinct`...

But yeah, I agree with formally disallowing `distinct object/tuple/enum`. It's possible to probably make it *slightly* more usable with the addition of the new `distinctBase` macro (which I have yet to play around with), but at that point, you just have to ask yourself: *why?*
